action/protocol/execution/evm/evmstatedbadapter.go

```diff
@@ -526,64 +535,66 @@ func (stateDB *StateDBAdapter) GetCodeSize(evmAddr common.Address) int {
 // SetCode sets contract's code
 func (stateDB *StateDBAdapter) SetCode(evmAddr common.Address, code []byte) {
 	addr := hash.BytesToHash160(evmAddr[:])
-	if contract, ok := stateDB.cachedContract[addr]; ok {
-		contract.SetCode(hash.Hash256b(code), code)
-		return
-	}
 	contract, err := stateDB.getContract(addr)
```

Comment:
if contract, ok := stateDB.cachedContract[addr]; ok{} is merged into stateDB.getContract(), to return the Contract object, then call SetCode()

the same is done for SetState(), GetState(), this simplifies the code, and as a result, removes getContractState() and setContractState()